### PR TITLE
RSFB-2795: Subquery alias

### PR DESCRIPTION
**Pull request by nosheenSheikh02**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java [hunk 1]
```diff
@@ -716,7 +741,18 @@ public Result visit(Aggregate e) {
...
+//        && e.getRowType().getFieldNames().equals(e.getInput(0).getRowType().getFieldNames())
+//        && e.getInput(0).getTraitSet().getTrait(SubQueryAliasTraitDef.instance) != null) {
+//      String subQueryAlias = requireNonNull(e.getInput(0).getTraitSet().
+//          getTrait(SubQueryAliasTraitDef.instance)).getSubQueryAlias();
+//      RelDataType rowType = this.adjustedRowType(e, result.node);
+//      result = result(result.node, ImmutableList.of(Clause.SELECT),
+//          subQueryAlias, rowType, ImmutableMap.of(subQueryAlias, rowType));
+//      return result;
+//    }
+//    return result;
```
[the remark this answers is not in the document]
Comment (naveenantonie) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java]: removed

File: core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java [hunk 2]
```diff
@@ -1721,4 +1792,16 @@ private SqlNode updateSqlWithNode(SqlImplementor.Result result) {
     CTERelToSqlUtil.updateSqlNode(sqlSelect);
     return sqlSelect;
   }
+
+  Result adjustResultWithSubQueryAlias(RelNode e, Result result) {
+    SubQueryAliasTrait subQueryAliasTrait =
+        e.getTraitSet().getTrait(SubQueryAliasTraitDef.instance);
+    if (subQueryAliasTrait != null) {
+      String subQueryAlias = subQueryAliasTrait.getSubQueryAlias();
+      RelDataType rowType = this.adjustedRowType(e, result.node);
```
[the remark this answers is not in the document]
Comment (naveenantonie) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java]: removed this keyword

File: core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java [hunk 3]
```diff
@@ -1721,4 +1792,16 @@ private SqlNode updateSqlWithNode(SqlImplementor.Result result) {
...
+      result =
+          result(result.node, result.clauses, subQueryAlias, rowType, ImmutableMap.of(subQueryAlias, rowType));
```
[the remark this answers is not in the document]
Comment (naveenantonie) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java]: adjusted

File: core/src/main/java/org/apache/calcite/rel/rules/dm/SerializeDistinctStructRule.java [hunk 4]
```diff
@@ -51,29 +56,36 @@ public void setExtension(RuleMatchExtension extension) {
...
+              .anyInputs())
+          .as(Config.class);
     }
 
     static boolean isDistinctStructExists(Aggregate e) {
       DistinctTrait distinctTrait = e.getTraitSet().getTrait(DistinctTraitDef.instance);
-      if (distinctTrait != null && distinctTrait.isDistinct() && !distinctTrait.isEvaluated()) {
+      SubQueryAliasTrait subQueryAliasTrait =
+          e.getTraitSet().getTrait(SubQueryAliasTraitDef.instance);
+      if (subQueryAliasTrait == null && distinctTrait != null
```
[the remark this answers is not in the document]
Comment (naveenantonie) [reply core/src/main/java/org/apache/calcite/rel/rules/dm/SerializeDistinctStructRule.java]: simplified as told

**Comment by vikassuryawanshi1** [comment]
Looks good to me

**Review by dm-shantanuc — APPROVED** [review]
Approved

**Pull request merged**